operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/dependent/EventSourceAware.java

```diff
@@ -1,10 +1,36 @@
 package io.javaoperatorsdk.operator.api.reconciler.dependent;
 
+import java.util.Optional;
+
 import io.fabric8.kubernetes.api.model.HasMetadata;
-import io.javaoperatorsdk.operator.processing.event.EventSourceRetriever;
+import io.javaoperatorsdk.operator.api.reconciler.EventSourceContext;
+import io.javaoperatorsdk.operator.processing.event.source.ResourceEventSource;
+
+public interface EventSourceAware<R, P extends HasMetadata> {
```

Comment:
A couple of remarks with this:

- This is in one hand an elegant solution to couple the event source related methods into one interface
on the other hand one of the primary reasons was to put these methods to Dependent Resource to couple it more, thus to express that a dependent resource ideally (an usually, so if not it is more of an corner case) should provide an event source. With this we are getting back to the position that basically event source is an optional thing in general.

- We don't have parity with the `@Dependent` annotation, so while setting there the `eventSource` parameter won't have effect on the dependent if it does not implement this interface. (Not that big deal just pointing out).

Reply:
Yes, on both. This was an initial attempt to see where that could lead. We can (and probably should) refine things.